### PR TITLE
Swift 3 support

### PR DESCRIPTION
**Pull request by eliperkins**
This brings in the work that @soffes and @n8armstrong have done on getting the Swift 3 branch going.

Supercedes #60 which seems to have become stale.

### CODE REVIEW

**Comment by eliperkins** [comment]
@hyperspacemark @soffes would you mind reviewing this?

**Comment by eliperkins** [comment]
Also, mfw I'm not a maintainer of this project because I'm not part of the @venmo org anymore.

File: DVR/Tests/SessionUploadTests.swift
Comment (lionel-alves) [inline DVR/Tests/SessionUploadTests.swift]: ⬅️

File: DVR/HTTPURLResponse.swift
Comment (lionel-alves) [inline DVR/HTTPURLResponse.swift:4]: Naming this class `HTTPURLResponse` requires us to use `Foundation.HTTPURLResponse` in a lot of places here and could be confusing. What do you think about `MutableHTTPURLResponse` since it's a mutable version of `HTTPURLResponse`?

File: DVR/HTTPURLResponse.swift
Comment (lionel-alves) [inline DVR/HTTPURLResponse.swift]: ⬅️

File: DVR/Session.swift
Comment (lionel-alves) [inline DVR/Session.swift]: ⬅️

File: DVR/Session.swift
Comment (lionel-alves) [inline DVR/Session.swift]: ⬅️

File: DVR/Session.swift
Comment (lionel-alves) [inline DVR/Session.swift]: ⬅️

File: DVR/Session.swift
Comment (lionel-alves) [inline DVR/Session.swift]: ⬅️

File: DVR/Session.swift
Comment (lionel-alves) [inline DVR/Session.swift]: ⬅️

**Review by lionel-alves — APPROVED** [review]
Thanks for getting that going Eli!

One naming suggestion but would love to have this merged 🍏

**Comment by soffes** [comment]
Thanks for owning this @eliperkins <3

File: DVR/HTTPURLResponse.swift
Comment (hyperspacemark) [reply DVR/HTTPURLResponse.swift:4]: I had a similar concern but since this isn't a part of the public API, I think we can address it later and not block merging this.

**Pull request merged**
